metabase/metabase
# Allow setting labels for filters

Closes https://github.com/metabase/metabase/issues/44289

[Testing plan lives here](https://www.notion.so/metabase/test-plan-Add-custom-list-input-support-and-labels-for-parameters-111e1ea897914b7587a35bfaa5c8b24f)

### Description

Allow setting custom labels for dashboard parameters.

### To test

1. Create a dashboard with a visualization for the Accounts table
2. Create a text parameter and link it to Accounts -> First Name
3. Edit the parameter, set "How should people filter on this column?" to "Search box" and click "Edit"
4. In the modal select "Custom list" and enter
  ```
  Gaston, Leo
  Ben, Benjamin
  Kasey
  ```
5. Click Done and save the dashboard
6. Select the parameter and in the dropdown input enter "Leo", the corresponding option should show up, select it. The value tag should show "Leo"
7. Click "Update filter"
9. The paramter widget should show "Leo"
10. The dashboard should be filtered for "Gaston" values


Do the same when setting "How should people filter on this column?" to  "Dropdown list" and make sure Leo shows up

## Demo
https://github.com/metabase/metabase/assets/1250185/dcf5b34d-2de2-4d67-84ce-235e248f2cff
